src/main/java/edu/byu/cs/autograder/compile/ModuleIndependenceVerifier.java

```diff
@@ -5,53 +5,66 @@
 
 import java.io.File;
 import java.io.IOException;
-import java.nio.file.Path;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
 import java.util.stream.Collectors;
 
-public class ModuleIndependenceVerifier implements StudentCodeInteractor {
+public class ModuleIndependenceVerifier implements StudentCodeVerifier {
     @Override
-    public void interact(GradingContext context, StudentCodeReader reader) throws GradingException {
-        Set<File> serverFiles = reader.filesMatching(".*server/src/main/java/.*").collect(Collectors.toSet());
-        Path serverPath = context.stageRepo().toPath().resolve("server/src/main/java");
-        Set<String> serverClasses = qualifiedNames(serverFiles, serverPath);
+    public void verify(GradingContext context, StudentCodeReader reader) throws GradingException {
+        try {
+            Set<File> serverFiles = reader.filesMatching(".*server/src/main/java/.*").collect(Collectors.toSet());
+            Set<String> serverPackages = packageNames(serverFiles, reader);
 
-        Set<File> clientFiles = reader.filesMatching(".*client/src/main/java/.*").collect(Collectors.toSet());
-        Path clientPath = context.stageRepo().toPath().resolve("client/src/main/java");
-        Set<String> clientClasses = qualifiedNames(clientFiles, clientPath);
+            Set<File> clientFiles = reader.filesMatching(".*client/src/main/java/.*").collect(Collectors.toSet());
+            Set<String> clientPackages = packageNames(clientFiles, reader);
 
-        try {
-            checkImports(context, reader, serverFiles, clientClasses);
-            checkImports(context, reader, clientFiles, serverClasses);
+            checkImports(context, reader, serverFiles, clientPackages);
+            checkImports(context, reader, clientFiles, serverPackages);
         } catch (IOException e) {
             throw new GradingException("Unable to verify module independence", e);
         }
     }
 
-    private Set<String> qualifiedNames(Set<File> files, Path relativeTo) {
-        Set<String> classes = new HashSet<>();
+    private Set<String> packageNames(Set<File> files, StudentCodeReader reader) throws IOException {
+        Set<String> packages = new HashSet<>();
         for(File file : files) {
-            String fileName = relativeTo.relativize(file.toPath()).toString();
-            if(!fileName.contains(".java")) continue;
-            classes.add(fileName.substring(0, fileName.indexOf('.')).replace('/', '.'));
+            if(!file.getName().contains(".java")) continue;
```

Comment:
Did you mean to use `.endsWith(".java");` instead of `.contains()`? What examples of files would you want to include that contain ".java" in the middle, but DO NOT end with ".java"?

Reply:
Yes. Thank you